spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/NaiveBayesClassifier.java

```diff
@@ -0,0 +1,157 @@
+      this.fieldToClassify = fieldToClassify;
+      this.fieldToSet = fieldToSet;
+    }
+  }
+
+  // for unit tests, otherwise config is injected by plugin framework.
+  public NaiveBayesClassifier(Config config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    Schema inputSchema = stageConfigurer.getInputSchema();
+    // if null, the input schema is unknown, or its multiple schemas.
+    if (inputSchema == null) {
+      outputSchema = null;
+      stageConfigurer.setOutputSchema(null);
+      return;
+    }
+
+    // otherwise, we have a constant input schema. Get the input schema and
+    // add a field to it, on which the prediction will be set
+    outputSchema = getOutputSchema(inputSchema);
+    stageConfigurer.setOutputSchema(outputSchema);
+  }
+
+  private Schema getOutputSchema(Schema inputSchema) {
+    // TODO: check if the fieldToSet is already set on the input?
+    List<Schema.Field> fields = new ArrayList<>(inputSchema.getFields());
+    fields.add(Schema.Field.of(config.fieldToSet, Schema.of(Schema.Type.DOUBLE)));
+    return Schema.recordOf(inputSchema.getRecordName() + ".predicted", fields);
+  }
+
+
+  @Override
+  public JavaRDD<StructuredRecord> transform(SparkPluginContext context,
+                                             JavaRDD<StructuredRecord> input) throws Exception {
+    FileSet fileSet = context.getDataset(config.fileSetName);
+    Location modelLocation = fileSet.getBaseLocation().append(config.path);
+    if (!modelLocation.exists()) {
+      LOG.warn("Failed to find model to use for classification. Location does not exist: {}.", modelLocation);
+      return input;
+    }
+
+    // load the model from a file in the model fileset
+    JavaSparkContext javaSparkContext = context.getOriginalSparkContext();
+    SparkContext sparkContext = JavaSparkContext.toSparkContext(javaSparkContext);
+    final NaiveBayesModel loadedModel = NaiveBayesModel.load(sparkContext, modelLocation.toURI().getPath());
+
+    final HashingTF tf = new HashingTF(100);
+
+    JavaRDD<StructuredRecord> output = input.map(new Function<StructuredRecord, StructuredRecord>() {
+      @Override
+      public StructuredRecord call(StructuredRecord structuredRecord) throws Exception {
+        String text = structuredRecord.get(config.fieldToClassify);
+        Vector vector = tf.transform(Lists.newArrayList(text.split(" ")));
+        double prediction = loadedModel.predict(vector);
+
+        return cloneRecord(structuredRecord)
+          .set(config.fieldToSet, prediction)
+          .build();
+      }
+    });
+    return output;
+  }
+
+  // creates a builder based off the given record
+  private StructuredRecord.Builder cloneRecord(StructuredRecord record) {
+    Schema schemaToUse = outputSchema != null ? outputSchema : record.getSchema();
```

Comment:
err, this is wrong. I need to not directly use the record.getSchema, but add a field to it first.